### PR TITLE
Fixed Windows build issues and made it independent on system wide installs

### PR DESCRIPTION
**Pull request by SeleDreams**
I fixed issues preventing it to build on windows and made it so it uses hts_engine as a submodule rather than as a system wide install so that people don't need to install anything before to be able to build it on their OS

### CODE REVIEW

**Comment by SeleDreams** [comment]
it seems like your CI isn't downloading the submodules which causes this fail normally it should be cloned with git clone --recursive

**Comment by SeleDreams** [comment]
I fixed the CI normally

**Comment by SeleDreams** [comment]
I think making the CI generate artifacts could be a good idea but it depends on what you prefer

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -16,12 +16,8 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR}/lib)
...
 # find HTSEngine
 # adapted from https://github.com/r9y9/open_jtalk/blob/1.10/src/CMakeLists.txt
-find_path(HTS_ENGINE_INCLUDE_DIR HTS_engine.h)
-find_library(HTS_ENGINE_LIB hts_engine_API)
-if(NOT HTS_ENGINE_LIB)
-  # fallback
-  find_library(HTS_ENGINE_LIB HTSEngine)
-endif()
+
+add_subdirectory(lib/hts_engine_API/hts_engine/src)
```
Comment (r9y9) [inline src/CMakeLists.txt:20]: Is it possible to revert this change? It would be useful to have it as a shared library so that the other libraries (such as open_jtalk) can use it as well.

Comment (r9y9) [reply src/CMakeLists.txt:20]: Or adding an option to embed hts_engine_API source as well.

**Review by r9y9 — COMMENTED** [review]
WIndows fixes look good to me, thanks! I would appreciate it if you could consider reverting the changes for including hts_engine_API as a submodule.

File: src/CMakeLists.txt (hunk 1 above)
Comment (SeleDreams) [reply src/CMakeLists.txt:20]: I think the issue is that Windows doesn't have a native package manager like linux has so it's a bit harder to handle global dependencies, if open_jtalk uses hts_engine, i think it could also use it as a submodule

Comment (SeleDreams) [reply src/CMakeLists.txt:20]: what do you think would be best ?

Comment (r9y9) [reply src/CMakeLists.txt:20]: I am so sorry for the late reply. I was wondering if we want to make the static build a configurable option. That said, I am okay with the change. if you are okay I will merge this PR.

Comment (SeleDreams) [reply src/CMakeLists.txt:20]: actually it is totally possible to make hts_engine_api a shared library instead of a static library

Comment (r9y9) [reply src/CMakeLists.txt:20]: Thanks. I am gonna merge ths PR!

**Review by r9y9 — APPROVED** [review]
